File: frescophotoview/src/main/java/com/zml/frescophotoview/ScaleFactorRetriever.java
package com.zml.frescophotoview;

/**
 * @autrhor zhangminglei01
 * @date 2021/6/7 11:06 上午
 */
public interface ScaleFactorRetriever {
    //返回值需>=1
    float nextFactor(float currFactor);
}

File: frescophotoview/src/main/java/com/zml/frescophotoview/ScaleFactorRetriever.kt
package com.zml.frescophotoview

/**
 * @autrhor zhangminglei01
 * 每次双击缩放时可以通过此接口告知缩放大小
 */
interface ScaleFactorRetriever {
    /**
     * @param currFactor 当前的缩放参数因子
     * @return 下一个缩放参数因子
     */
    fun nextFactor(currFactor: Float): Float
}